api/app/controllers/spree/api/v1/orders_controller.rb

```diff
@@ -121,7 +121,7 @@ def normalize_params
           end
 
           def find_order(lock = false)
-            @order = Spree::Order.lock(lock).friendly.find(params[:id])
+            @order = Spree::Order.lock(lock).find_by!(number: params[:id])
```

Comment:
@krzysiek1507 all API calls are number based, that's what official guides enforced years ago